client/state/media/reducer.js

```diff
@@ -2,40 +2,79 @@
  * External dependencies
  */
 import { combineReducers } from 'redux';
-import { keyBy, omit } from 'lodash';
+import { omit } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { MEDIA_DELETE, MEDIA_RECEIVE } from 'state/action-types';
+import {
+	MEDIA_DELETE,
+	MEDIA_RECEIVE,
+	MEDIA_REQUEST_FAILURE,
+	MEDIA_REQUESTING } from 'state/action-types';
 import { createReducer } from 'state/utils';
+import MediaQueryManager from 'lib/query-manager/media';
 
-export const items = createReducer( {}, {
-	[ MEDIA_RECEIVE ]: ( state, action ) => {
-		const { siteId, media } = action;
+export const queries = ( () => {
+	function applyToManager( state, siteId, method, createDefault, ...args ) {
+		if ( ! state[ siteId ] ) {
+			if ( ! createDefault ) {
+				return state;
+			}
+
+			return {
+				...state,
+				[ siteId ]: ( new MediaQueryManager() )[ method ]( ...args )
+			};
+		}
+
+		const nextManager = state[ siteId ][ method ]( ...args );
+
+		if ( nextManager === state[ siteId ] ) {
+			return state;
+		}
+
+		return {
+			...state,
+			[ siteId ]: nextManager
+		};
+	}
+
+	return createReducer( {}, {
+		[ MEDIA_RECEIVE ]: ( state, { siteId, media, found, query } ) => {
+			return applyToManager( state, siteId, 'receive', true, media, { found, query } );
+		},
+		[ MEDIA_DELETE ]: ( state, { siteId, mediaIds } ) => {
+			return applyToManager( state, siteId, 'removeItems', true, mediaIds );
+		}
+	} );
+} )();
```

Comment:
This IIFE is a bit curious to me here. Can you clarify what it's being used for?